src/components/Modal/BaseModal.tsx

```diff
@@ -18,10 +19,26 @@ import useNativeDriver from '@libs/useNativeDriver';
 import variables from '@styles/variables';
 import * as Modal from '@userActions/Modal';
 import CONST from '@src/CONST';
+import BottomDockedModal from './BottomDockedModal';
+import type ModalProps from './BottomDockedModal/types';
 import ModalContent from './ModalContent';
 import ModalContext from './ModalContext';
 import type BaseModalProps from './types';
 
+type ModalComponentProps = (ReactNativeModalProps | ModalProps) & {
+    type?: ValueOf<typeof CONST.MODAL.MODAL_TYPE>;
+    shouldUseNewModal: boolean;
+};
+
+function ModalComponent({type, shouldUseNewModal, ...props}: ModalComponentProps) {
+    if (type === CONST.MODAL.MODAL_TYPE.BOTTOM_DOCKED && shouldUseNewModal) {
```

Comment:
If we control all the modals with `shouldUseNewModal` then is `type` comparison needed too?